src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,207 @@
+            String[] arr = str.split(" ", 2);
+
+            String upperCaseCmd = Parser.parseCmd(arr[0]);
+            Command cmd;
+            try {
+                cmd = Command.valueOf(upperCaseCmd);
+            } catch (IllegalArgumentException ex) {
+                ui.invalidInput();
+                continue;
+            }
+
+            switch (cmd) {
+            case BYE:
+                ui.showBye();
+                // rewrite the file to update latest changes
+                Storage.saveFile(this.storage.getFile(), ls);
+                break WhileLoop;
+            case LIST:
+                if (ls.isEmpty()) {
+                    ui.showListNoTask();
+                } else {
+                    ui.showListTask();
+                    for (int i = 0; i < ls.size(); i++) {
+                        Task task = ls.get(i);
+                        int num = i + 1;
+                        System.out.println(num + ". " + task.toString());
+                    }
+                    ui.horizontalDiv();
+                }
+                break;
+            case DONE:
+                try {
+                    int numToBeMarkedAsDone = Parser.parseInt(str);
+                    Task tsk = ls.get(numToBeMarkedAsDone);
+                    tsk.markAsDone();
+                    ls.set(numToBeMarkedAsDone, tsk);
+                    ui.showDoneMsg(tsk.toString());
+                } catch (Exception ex) {
+                    ui.showDoneError();
+                }
+                break;
+            case TODO:
+                try {
+                    Task newTask = new Todo(arr[1]);
+                    ls.add(newTask);
+                    ui.showTodoMsg(ls, newTask);
+                } catch (Exception ex) {
+                    ui.showTodoError();
+                }
+                break;
+            case DEADLINE:
+                try {
+                    String[] arrOfStr = Parser.parse(arr, 1);
+                    try {
+                        Task newTask = new Deadline(arrOfStr[0], arrOfStr[1]);
+                        ls.add(newTask);
+                        ui.showDeadlineEventMsg(ls, newTask);
+                    } catch (Exception ex) {
+                        ui.showDeadlineFormatError();
+                    }
+                } catch (Exception ex) {
+                    ui.showDeadlineError();
+                }
+                break;
+            case EVENT:
+                try {
+                    String[] arrOfStr = Parser.parse(arr, 2);
+                    try {
+                        Task newTask = new Event(arrOfStr[0], arrOfStr[1]);
+                        ls.add(newTask);
+                        ui.showDeadlineEventMsg(ls, newTask);
+                    } catch (Exception ex) {
+                        ui.showEventFormatError();
+                    }
+                } catch (Exception ex) {
+                    ui.showEventError();
+                }
```

Comment:
You could try creating custom exception in which case you would not have to do a nested try and catch and instead you would get to do something like 
```try {
    Your code
} catch (illegalFormatException ex) {
     System.out.println(Error Message);
} catch (illegalTaskException ex) {
     System.out.println(Error message);
}```